server-spi-private/src/main/java/org/keycloak/models/locking/GlobalLockProvider.java

```diff
@@ -17,57 +17,73 @@
 
 package org.keycloak.models.locking;
 
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.models.KeycloakSessionTaskWithResult;
 import org.keycloak.provider.Provider;
 
 import java.time.Duration;
 
 public interface GlobalLockProvider extends Provider {
 
+    class Constants {
+        public static final String KEYCLOAK_BOOT = "keycloak-boot";
+    }
+
     /**
-     * Effectively the same as {@code acquire(lockName, null)}
-     * <p />
-     * This method is intended to be used in a {@code try}-with-resources block.
+     * Effectively the same as {@code withLock(lockName, null, task)}
```

Comment:
The first sentence (up until the first dot) should summarize what the method does, and only then continue to go into details.

```suggestion
     * Acquires a new non-reentrant global lock that is visible to all Keycloak nodes.
     * Effectively the same as {@code withLock(lockName, null, task)}.
```